src/views/Game.vue

```diff
@@ -51,30 +51,26 @@ export default {
   },
   firestore() {
     return {
-      detectives: db
+      players: db
         .collection("games")
         .doc(this.gameId)
-        .collection("detectives"),
+        .collection("players"),
       game: db.collection("games").doc(this.gameId),
     };
   },
   computed: {
     isMrX: function() {
-      const loggedInUserEmail = this.state.user && this.state.user.email;
-      return (
-        this.game &&
-        this.game["mr-x"] &&
-        this.game["mr-x"].user === loggedInUserEmail
-      );
+      const userId = this.state.user && this.state.user.uid;
+      return userId && this.mrX && this.mrX.user === userId;
+    },
+    detectives: function() {
+      return this.players.filter(player => player.role !== "mr-x");
     },
     roundNumber: function() {
       return this.game && this.game.round;
     },
     mrX: function() {
-      return this.game && this.game["mr-x"];
-    },
-    mrXPlayer: function() {
-      return this.game && this.game["mr-x"] && this.game["mr-x"].user;
+      return this.players.find(player => player.role === "mr-x");
```

Comment:
this is cleaner to me, rather than keeping Mr. X data on the game level

Reply:
Fully agree